### PR TITLE
[ray-data] Add alias parameters to the aggregate function, and add quantile fn

### PR DESCRIPTION
**Pull request by yiwei00000**
Hi:
I added the alias parameter in the aggregate function and returned the alias. At the same time, I added aggregate function, quantile .

The following is an example of usage :

from ray.data.aggregate import Max, Mean, Min, Quantile, Std
import ray

ls = [
    {"A": i, "B": i ** 2}
    for i in range(10)]
ds = ray.data.from_items(ls)

ops = [Max("A", alias_name="max_a"), Min("A", alias_name="min_a"), Mean("A", alias_name="mean_a"),
       Std("A", alias_name="std_a"),Quantile("A", percent=0.5, alias_name="quantile_a")]
print(dict(ds.aggregate(*ops)))

rs:
{'max_a': 9, 'min_a': 0, 'mean_a': 4.5, 'std_a': 3.0276503540974917, 'quantile_a': 4.5}

### CODE REVIEW

File: python/ray/data/aggregate.py [hunk 1]
```diff
@@ -312,3 +343,79 @@ def _to_on_fn(on: Optional[KeyFn]):
...
+            merge=null_merge,
+            accumulate_block=_null_wrap_accumulate_block(
+                ignore_nulls,
+                block_row_ls,
+                null_merge,
+            ),
+            finalize=_null_wrap_finalize(percentile),
+            name=(rs_name),
+        )
+
```
Comment (ericl) [inline python/ray/data/aggregate.py:416]: Some lint issues (run scripts/format.sh to fix): https://buildkite.com/ray-project/oss-ci-build-pr/builds/18407#01877969-e28d-42df-b046-0e7a516a22a5

File: python/ray/data/aggregate.py [hunk 2]
```diff
@@ -113,11 +113,16 @@ def __init__(self):
 class Sum(_AggregateOnKeyBase):
     """Defines sum aggregation."""
 
-    def __init__(self, on: Optional[KeyFn] = None, ignore_nulls: bool = True):
+    def __init__(self, on: Optional[KeyFn] = None, ignore_nulls: bool = True, alias_name: Optional[KeyFn] = None):
```
Comment (ericl) [inline python/ray/data/aggregate.py:116]: Could you also add unit tests for this feature in test_dataset_all_to_all.py ?

**Review by ericl — COMMENTED** [review]
API +1 from me, cc @c21 @scottjlee

File: python/ray/data/aggregate.py (hunk 1 above)
Comment (yiwei00000) [reply python/ray/data/aggregate.py:416]: Sure, I will execute this sh immediately. Thank you very much

File: python/ray/data/aggregate.py (hunk 2 above)
Comment (yiwei00000) [reply python/ray/data/aggregate.py]: Yes, I will add test case in this file

Comment (yiwei00000) [reply python/ray/data/aggregate.py]: Thanks again, test case has been added to test_ dataset_ all_ to_ All files are included

File: python/ray/data/aggregate.py [hunk 3]
```diff
@@ -312,3 +368,79 @@ def _to_on_fn(on: Optional[KeyFn]):
...
+
+
+@PublicAPI
+class Quantile(_AggregateOnKeyBase):
+    """Defines Quantile aggregation."""
+
+    def __init__(
+        self,
+        on: Optional[KeyFn] = None,
+        percent: float = 0.5,
```
Comment (scottjlee) [inline python/ray/data/aggregate.py:380]: other libraries such as [pandas](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.quantile.html) and [numpy](https://numpy.org/doc/stable/reference/generated/numpy.quantile.html) use the parameter name `q` for quantiles -- should we also use this notation convention?

**Review by scottjlee — APPROVED** [review]
LGTM, just a couple small nits. Thanks!

File: python/ray/data/aggregate.py [hunk 4]
```diff
@@ -312,3 +368,79 @@ def _to_on_fn(on: Optional[KeyFn]):
...
+            return ls
+
+        if alias_name:
+            rs_name = alias_name
+        else:
+            rs_name = f"quantile({str(on)})"
+
+        import math
+
+        def percentile(N, key=lambda x: x):
```
[the remark this answers is not in the document]
Comment (yiwei00000) [reply python/ray/data/aggregate.py]: I have updated it, thank you!!!

**Comment by yiwei00000** [comment]
Hi，is there any other issues? I will cooperate and correct it in a timely manner.   @scottjlee @ericl @scv119

**Comment by ericl** [comment]
Lgtm. I think there's one unresolved comment from Scott above still.

File: python/ray/data/aggregate.py (hunk 3 above)
Comment (yiwei00000) [reply python/ray/data/aggregate.py]: Thank you. I have changed the percentage parameter to q, just like Pandas and Numpy.

**Comment by ericl** [comment]
Looks like there's also a merge conflict with master

**Comment by yiwei00000** [comment]
> Looks like there's also a merge conflict with master

**Comment by yiwei00000** [comment]
> Looks like there's also a merge conflict with master

Yes, I don't know if there is a file name conflict. I merged the upstream/master and the conflict was resolved. @ericl

**Comment by ericl** [comment]
Merged, thanks!

**Comment by yiwei00000** [comment]
> Merged, thanks!

Thank you very much. I still have some areas to add or modify in the future, including APIs or new features.

**Pull request merged**